### PR TITLE
CSB-3797 Add overrides for dependency convergence

### PR DESCRIPTION
**Pull request by cunningt**
https://issues.redhat.com/browse/CSB-3797

Added a number of overrides for dependency convergence.     

The requested overrides are here : https://docs.google.com/document/d/1byV3eWepcnBSdGNqgCfxaLVvOAU8Obdyv-0qUk7io_A/edit

### CODE REVIEW

File: tooling/camel-spring-boot-dependencies/pom.xml
Comment (Croway) [inline tooling/camel-spring-boot-dependencies/pom.xml:1]: Shouldn't the generator be updated as well? https://github.com/jboss-fuse/camel-spring-boot/blob/d6ecc5920d620f91c8efa23f6c7032cf2b672228/tooling/camel-spring-boot-dependencies-generator/pom.xml

Comment (cunningt) [reply tooling/camel-spring-boot-dependencies/pom.xml:1]: I don't think that's necessary - the overrides that have been added come before org.apache.camel.springboot:camel-spring-boot-bom, so they take precedence.

Comment (Croway) [reply tooling/camel-spring-boot-dependencies/pom.xml:1]: I am a bit lost, https://github.com/jboss-fuse/camel-spring-boot/pull/352/files#diff-348c97be06ad41e16f1a343737bfc6752a28e663b06621875a6fb2a76da1327fR36 in this way you need to fix the version, and the variable cannot be used.
 
 Using the generator, can't we use the properties? and then, the BOM with the versions is generated during the build.

Comment (luigidemasi) [reply tooling/camel-spring-boot-dependencies/pom.xml:1]: @cunningt is this pom generated by camel-spring-boot-dependencies-generator? if yes than probably @Croway is right, we should put the changes there.

Comment (cunningt) [reply tooling/camel-spring-boot-dependencies/pom.xml:1]: @Croway @luigidemasi Talked to Federico on slack about this - it's confusing to me too but the camel-spring-boot-dependencies entries here are not hardcoded, they are being generated from the dependencyManagement entries in camel-spring-boot/pom.xml.    The dependencyManagement entries in the root pom.xml are necessary for keeping those aligned with productized versions of aalto/kryo/jakarta.el/etc - the only place we are setting this version property is here : https://github.com/jboss-fuse/camel-spring-boot/pull/352/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R141

**Review by Croway — APPROVED** [review]

**Review by luigidemasi — APPROVED** [review]

**Pull request merged**